code/modules/antagonists/heretic/items/heretic_necks.dm

```diff
@@ -0,0 +1,166 @@
+	icon_state = "crimson_medallion"
+	/// The aura healing component. Used to delete it when taken off.
+	var/datum/component/component
+	/// If active or not, used to add and remove its cult and heretic buffs.
+	var/active = FALSE
+
+/obj/item/clothing/neck/heretic_focus/crimson_medallion/equipped(mob/living/user, slot)
+	. = ..()
+	if(!(slot & ITEM_SLOT_NECK))
+		return
+
+	if(IS_CULTIST(user))
+		var/datum/action/innate/cult/blood_magic/magic_holder = locate() in user.actions
+		magic_holder.magic_enhanced = TRUE
+	else if(IS_HERETIC_OR_MONSTER(user) && !active)
+		for(var/datum/spell/spell_action in user.mob_spell_list)
+			spell_action.cooldown_handler.recharge_duration *= 0.5
+			active = TRUE
+
+	ADD_TRAIT(user, TRAIT_MANSUS_TOUCHED, UID(src))
+	to_chat(user, "<span class='alert'>Your heart takes on a strange yet soothing irregular rhythm, and your blood feels significantly less viscous than it used to be. You're not sure if that's a good thing.</span>")
+	// AOE healing here
+
+/obj/item/clothing/neck/heretic_focus/crimson_medallion/dropped(mob/living/user)
+	. = ..()
+
+	if(!istype(user))
+		return
+
+	if(HAS_TRAIT_FROM(user, TRAIT_MANSUS_TOUCHED, UID(src)))
+		to_chat(user, "<span class='notice'>Your heart and blood return to their regular old rhythm and flow.</span>")
+
+	if(IS_HERETIC_OR_MONSTER(user) && active)
+		for(var/datum/spell/spell_action in user.mob_spell_list)
+			spell_action.cooldown_handler.recharge_duration *= 2
+			active = FALSE
+	QDEL_NULL(component)
+	REMOVE_TRAIT(user, TRAIT_MANSUS_TOUCHED, UID(src))
+
+	// If boosted enable is set, to prevent false dropped() calls from repeatedly nuking the max spells.
+	var/datum/action/innate/cult/blood_magic/magic_holder = locate() in user.actions
+	// Remove the last spell if over new limit, as we will reduce our max spell amount. Done beforehand as it causes a index out of bounds runtime otherwise.
+	if(magic_holder?.magic_enhanced)
+		QDEL_NULL(magic_holder.spells[ENHANCED_BLOODCHARGE])
+	magic_holder?.magic_enhanced = FALSE
+
+
+/obj/item/clothing/neck/heretic_focus/crimson_medallion/activate_self(mob/user)
+	if(..() || !isliving(user))
+		return
+	var/mob/living/our_user = user
+	to_chat(our_user, "<span class='danger'>You start tightly squeezing [src]...</span>")
+	if(!do_after(user, 1.25 SECONDS, src))
+		return
+	to_chat(our_user, "<span class='danger'>[src] explodes into a shower of gore and blood, drenching your arm. You can feel the blood seeping into your skin. You inmediately feel better, but soon, the feeling turns hollow as your veins itch.</span>")
+	new /obj/effect/gibspawner/generic(get_turf(src))
+	var/heal_amt = our_user.adjustBruteLoss(-50)
+	our_user.adjustFireLoss( -(50 - abs(heal_amt)) ) // no double dipping
+
+	// I want it to poison the user but I also think it'd be neat if they got their juice as well. But that cancels most of the damage out. So I dunno.
```

Comment:
feels loosely like a todo

Reply:
I'll remove it, was in the file when ported